src/lib/style-properties.html

```diff
@@ -121,63 +150,150 @@
         return parts.join(';');
       },
 
-      // TODO(srovell): make this go away by pre-processing rules
-      // to contain specifically only the css that consumes properties 
-      // for use here.
-      //
-      // returns cssText with properties applied;
-      // if no properties apply, returns nothing
       applyProperties: function(rule, props) {
-        var cssText = rule.parsedCssText;
         var output = '';
         // dynamically added sheets may not be decorated so ensure they are.
         if (!rule.properties) {
           this.decorateRule(rule);
         }
-        if (rule.properties.consumes) {
-          var m, v;
-          // cssText including {...} will confuse this function so we lim it
-          cssText = cssText.replace(this.rx.BRACKETED, '');
-          // e.g. color: var(--color);
-          while (m = this.rx.VAR_USE.exec(cssText)) {
-            v = this.valueForProperty(m[2], props);
-            if (v) {
-              output += '\t' + m[1].trim() + ': ' + this.propertyToCss(v);
-            }
+        if (rule.propertyInfo.cssText) {
+          output = this.valueForProperties(rule.propertyInfo.cssText, props);
+        }
+        rule.cssText = output;
+      },
+
+      // Test if the rules in these styles matche the given `element` and if so,
+      // collect any custom properties into `props`.
+      propertiesFromStyles: function(styles, element) {
+        var props = {}, self = this;
+        styleUtil.forRulesInStyles(styles, function(rule) {

```

Comment:
TODO: the rules list itself could be shorter (pre-filter on rules with properties)